pyramid/csrf.py

```diff
@@ -214,8 +218,37 @@ def check_csrf_token(request,
        to use the configured :class:`pyramid.interfaces.ICSRFStoragePolicy` to
        verify the CSRF token.
 
+    # TODO: add versionchanged and the new safe_methods parameter doc here
     """
     supplied_token = ""
+
+    # Try to read options from `config.set_default_csrf_options` set previously
+    # as the default value
+    csrf_options = request.registry.getUtility(IDefaultCSRFOptions)
+    if header is None:
+        if csrf_options.header is not None:
+            header = csrf_options.header
+        else:
+            # TODO: extract as constant var?
+            header = 'X-CSRF-Token'
+    if token is None:
+        if csrf_options.token is not None:
+            token = csrf_options.token
+        else:
+            # TODO: extract as constant var?
+            token = 'csrf_token'
+    if safe_methods is None:
+        if csrf_options.safe_methods is not None:
+            safe_methods = csrf_options.safe_methods
+        else:
+            # TODO: I saw these default values appear at many places, maybe we
+            # should extract them as constant values somehwere and use them,
+            # so that we won't be out of sync?
+            safe_methods = frozenset(["GET", "HEAD", "OPTIONS", "TRACE"])
```

Comment:
I think one good option is just to always define CSRF options by calling `config.set_default_csrf_options(require_csrf=False)` in `add_default_security` here:

https://github.com/Pylons/pyramid/blob/f6f6d13be6c9ca384fefc35729ecf32a20b8b50d/pyramid/config/security.py#L21

Then you can remove the duplication and just assume you can query a `IDefaultCSRFOptions` object from the registry.